src/tests/Interop/Swift/SwiftErrorHandling/SwiftErrorHandling.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Swift;
+using System.Text;
+using Xunit;
+
+public class ErrorHandlingTests
+{
+    private const string SwiftLib = "libSwiftErrorHandling.dylib";
+
+    [DllImport(SwiftLib, EntryPoint = "$s18SwiftErrorHandling05setMyB7Message5bytes6lengthySPys5UInt8VG_SitF")]
+    public static extern void SetErrorMessage(byte[] strBytes, int length);
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(SwiftLib, EntryPoint = "$s18SwiftErrorHandling018conditionallyThrowB004willE0SiSb_tKF")]
+    public unsafe static extern nint conditionallyThrowError(bool willThrow, SwiftError* error);
+
+    [DllImport(SwiftLib, EntryPoint = "$s18SwiftErrorHandling05getMyB7Message4fromSPys5UInt8VGSgs0B0_p_tF")]
+    public static extern IntPtr GetErrorMessage(IntPtr handle);
+
+    [Fact]
+    public unsafe static void TestSwiftErrorThrown()
+    {
+        const string expectedErrorMessage = "Catch me if you can!";
+        SetErrorMessageForSwift(expectedErrorMessage);
+
+        SwiftError error;
+
+        // This will throw an error
+        conditionallyThrowError(true, &error);
+        Assert.True(error.Value != IntPtr.Zero, "A Swift error was expected to be thrown.");
+
+        string errorMessage = GetErrorMessageFromSwift(error, expectedErrorMessage.Length);
```

Comment:
Should `GetErrorMessageFromSwift` be responsible for computing the error message length? The length of the actual message does not have to be `expectedErrorMessage.Length` when things go wrong.

If the error message is zero-terminated C string, you can use `MemoryMarshal.CreateReadOnlySpanFromNullTerminated` or ever just `Marshal.PtrToStringUTF8`.

Reply:
You are right. The `Marshal.PtrToStringUTF8` allocates a managed string and copies zero-terminated C string into it.